algorithms/polish.sbl

```diff
@@ -0,0 +1,178 @@
+stringescapes {}
+
+stringdef a' '{U+0105}' 
+stringdef e' '{U+0119}' 
+stringdef l' '{U+0142}' 
+stringdef s' '{U+015b}' 
+stringdef c' '{U+0107}' 
+stringdef o' '{U+00f3}' 
+
+
+
+
+externals (stem)
+
+
+routines (
+	remove_nouns
+	remove_dimunitive
+	remove_superlative_ends
+	remove_adjective_ends
+	remove_verbs_ends
+	remove_adverbs_ends
+	remove_plural_forms
+	remove_general_ends
+)
+
+
+backwardmode (
+	define remove_nouns as (
+		(
+			$(len > 7)
+			[substring] among (
+				'zacja' 'zacji'
```

Comment:
The corresponding original code was

```
('zacja' or 'zacja' or 'zacji')
```

which duplicates `'zacja'`.

I'm not familiar with the Polish language, but I notice the two other suffixes ending `cja` both have a corresponding suffix ending `cj{a'}` so I wonder if maybe one `'zacja'` here should have been `'zacj{a'}'`?

Reply:
Yes, you are right, `-zacją` is the accusative case of `-zacja`, probably it was meant here. Interestingly, adding it does not affect the tests, probably because all such words are also covered by the next case for `-acją`.

Reply:
Good spot, and it actually seems that first set of endings here is entirely redundant so might as well be removed - if `$(len > 7)` then `$(len > 6)` so we'll definitely do the next set and that does equivalent changes to everything in the first set.

If linguistically `-zacja`, etc are different endings we can just note in comments that `acja` handles both `-acja` and `-zacja`.

Reply:
Just to note now we know the origin of the algorithm, https://github.com/Tutanchamon/pl_stemmer/blob/master/pl_stemmer.py#L200 has `zacją` so this fixes a deviation from that:

```
    if len(word) > 7 and word[-5:] in {"zacja", u"zacją", "zacji"}:
```